GH-3280: NullChannel as reply for void gateways (#3292)
spring-projects/spring-integration: src/reference/asciidoc/gateway.adoc

You might also want to explicitly provide a reply channel for monitoring or auditing through an interceptor (for example, <<./channel.adoc#channel-wiretap,wiretap>>).
To configure a channel interceptor, you need a named channel.

NOTE: Starting with version 5.4, when gateway method return type is `void`, the framework populates a `replyChannel` header as a `nullChannel` bean reference if such a header is not provided explicitly.

> Such a behavior allows to ignore possible reply from downstream flow meeting a one-way gateway contract.

Reviewer: ```suggestion
This allows any possible reply from the downstream flow to be discarded, meeting the one-way gateway contract.
```
